crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs

```diff
@@ -66,6 +66,11 @@ pub enum SenderData {
         /// The user ID of the user who established this session.
         user_id: OwnedUserId,
 
+        /// The device ID of the device that send the session.
+        /// This is an `Option` for backwards compatibility, but we should
+        /// always populate it on creation.
+        device_id: Option<OwnedDeviceId>,
```

Comment:
@poljar Clippy is warning that `SenderData` now has `large size difference between variants` because `the largest variant contains at least 225 bytes` since I added the `device_id`.

I think this might be fine, since the `SenderKnown` variant is the "normal" one, so I am inclined to ignore this. What do you think? It would mean a user with lots of legacy sessions or unverified friends would be wasting memory.

Reply:
Can't you just box the type?

Reply:
Yes, do you think it's worth doing?

Reply:
Usually I blindly trust clippy, and boxing it is cheap, it's slightly more ugly than not boxing but I think we can live with that.

Reply:
Done in 7d3e49d798